ambry-store/src/test/java/com.github.ambry.store/CompactionManagerTest.java

```diff
@@ -228,17 +250,71 @@ public void testCompactionWithMisbehavingStores() throws Exception {
         assertTrue("Should schedule compaction", compactionManager.scheduleNextForCompaction(store));
         assertTrue("Compaction call did not come within the expected time",
             store.compactCallsCountdown.await(1, TimeUnit.HOURS));
+        if (store.callOrderException != null) {
+          throw store.callOrderException;
+        }
         assertTrue("compact() should have been called", store.compactCalled);
       }
     }
-
     compactionManager.disable();
     compactionManager.awaitTermination();
     assertFalse("Compaction thread should not be running", compactionManager.isCompactionExecutorRunning());
   }
 
+  /**
+   * Tests for cases where only certain triggers are enabled.
+   * @throws Exception
+   */
+  @Test
+  public void testDifferentTriggers() throws Exception {
+    properties.setProperty("store.enable.compaction", Boolean.toString(true));
+    blobStore.details = generateRandomCompactionDetails(2);
+
+    // all enabled is already tested in testCompactionExecutorHappyPath()
+    // only time based enabled
+    properties.setProperty("store.compaction.triggers", "Time");
```

Comment:
Would be good to have tests with 1) all the triggers enabled and 2) with none of them enabled.